sandbox/rocky/tf/algos/network/actor_critic_net.py

```diff
@@ -0,0 +1,220 @@
+import tensorflow as tf
+import tensorflow.contrib as tc
+import numpy as np
+
+
+def _fc(x,
+        output_dim,
+        weight_initializer=tf.contrib.layers.xavier_initializer(uniform=False),
+        bias_initializer=tf.zeros_initializer):
+    return tf.contrib.layers.fully_connected(x, output_dim, activation_fn=None, \
+            weights_initializer=weight_initializer, weights_regularizer=None, biases_initializer=bias_initializer)
+
+
+def _norm(x):
+    return tc.layers.layer_norm(x, center=True, scale=True)
+
+
+class ActorCriticBaseModel(object):
+    def __init__(self, name):
+        self.name = name
+
+    @property
+    def trainable_vars(self):
+        return tf.get_collection(
+            tf.GraphKeys.TRAINABLE_VARIABLES, scope=self.name)
+
+    def setup_target_net(self, target_net, tau):
+        vars = self.trainable_vars
+        target_vars = target_net.trainable_vars
+
+        soft_updates = []
+        vars = vars[0:len(target_vars)]
+        for var, target_var in zip(vars, target_vars):
+            soft_updates.append(
+                tf.assign(target_var, (1. - tau) * target_var + tau * var))
+        assert len(soft_updates) == len(vars)
+        self._update_paras = tf.group(*soft_updates)
+
+
+class ActorNet(ActorCriticBaseModel):
+    def __init__(self, sess, action_dim, lr=1e-4, bound=1):
+        self.name = 'actor'
+        self._action_dim = action_dim
+        self._lr = lr
+        self._session = sess
+        self._bound = bound
+
+    def build_net(self, state):
+        self._state = state
+        self._action = self._build_net(state) * self._bound
+
+        with tf.variable_scope(self.name) as scope:
+            self._optimizer = tf.train.AdamOptimizer(
+                self._lr, beta1=0.9, beta2=0.999, epsilon=1e-08)
+
+    def set_grad(self, grad):
+        grads = tf.gradients(
+            ys=self._action, xs=self.trainable_vars, grad_ys=grad)
+        self._train_op = self._optimizer.apply_gradients(
+            zip(grads, self.trainable_vars))
+
+    def _build_net(self, state):
```

Comment:
add support for user-defined number and sizes of hidden layers, e.g. `hidden_layers=[64, 64]`